src/main/java/graphql/schema/DataFetcherFactory.java

```diff
@@ -22,4 +22,17 @@ public interface DataFetcherFactory<T> {
      */
     DataFetcher<T> get(DataFetcherFactoryEnvironment environment);
 
+    /**
+     * Returns a {@link graphql.schema.DataFetcher} given the field definition
+     * which is cheaper in object allocation terms.
+     *
+     * @param fieldDefinition the field that needs the data fetcher
+     *
+     * @return a data fetcher
+     */
+
+    default DataFetcher<T> getViaField(GraphQLFieldDefinition fieldDefinition) {
+        return null;
+    }
```

Comment:
DataFetcherFactoryEnvironment is a "parameter" object that ONLYT wraps a GraphQLFieldDefinition

It was put in place for "future expansion reasons" but we never expanded it and never had to.  So this one as a bad bet since we have to make a new object just to pass 1 object we already have

This method undoes that and says we can lookup via just the GraphQLFieldDefinition object